### PR TITLE
Use session key for storing which question was asked.

### PR DESCRIPTION
**Pull request by mjfrey**
Use session attribute to store the question that was just asked so we can respond correctly. Also only speak the first 2 search results and ask the learner if they want to hear the rest.

### CODE REVIEW

**Comment by mjfrey** [comment]
@clintonb please take a look.  This may not be the exact wording but I think is the general idea.

File: enrollments/enrollments.py [hunk 1]
```diff
@@ -23,6 +23,7 @@
 
 QUESTION_KEY = 'QUESTION'
 ENROLLMENTS_KEY = 'ENROLLMENTS'
+SEARCH_KEY = "SEARCH"
```
Comment (clintonb) [inline enrollments/enrollments.py]: Nit: single quotes.

File: enrollments/enrollments.py [hunk 2]
```diff
@@ -203,11 +205,18 @@ def continue_interaction():
     if session.new:
         return help()
 
-    speech_text = '<speak><p>Your courses include</p>'
-    _enrollments = get_course_names(session.attributes[ENROLLMENTS_KEY])
+    if session.attributes[QUESTION_KEY] == 'enrollments':
```
Comment (clintonb) [inline enrollments/enrollments.py]: Extract `session.attributes[QUESTION_KEY]` to a variable.

File: enrollments/enrollments.py [hunk 3]
```diff
@@ -203,11 +205,18 @@ def continue_interaction():
...
+        speech_text = '<speak><p>Your courses include</p>'
+        _enrollments = get_course_names(session.attributes[ENROLLMENTS_KEY])
 
-    for enrollment in _enrollments.values():
-        speech_text += '<p>{}</p>'.format(enrollment['title'])
+        for enrollment in _enrollments.values():
+            speech_text += '<p>{}</p>'.format(enrollment['title'])
+
+    if session.attributes[QUESTION_KEY] == 'search':
```
Comment (clintonb) [inline enrollments/enrollments.py]: `elif`

File: enrollments/enrollments.py [hunk 4]
```diff
@@ -203,11 +205,18 @@ def continue_interaction():
...
+        courses = session.attributes[SEARCH_KEY]
+        speech_text = '<speak><p>The rest of the results are</p>'
```
Comment (clintonb) [inline enrollments/enrollments.py]: Add a space to avoid word munging.

File: enrollments/enrollments.py [hunk 5]
```diff
@@ -267,16 +276,23 @@ def search(subject):
...
+        for course in courses[:2]:
             speech_text += '<p>{}</p>'.format(course['title'])
 
-        speech_text += '</speak>'
+        if len(courses) == 2:
+            speech_text += '</speak>'
+            return statement(speech_text).simple_card(APP_NAME, speech_text)
+        else:
+            speech_text += '<p>Would you like to hear the rest?</p></speak>'
+            reprompt = 'Would you like to hear the rest of the results'
```
Comment (clintonb) [inline enrollments/enrollments.py]: Add a question mark. Punctuation affects inflection.

File: enrollments/enrollments.py [hunk 6]
```diff
@@ -267,16 +276,23 @@ def search(subject):
     if count <= 0:
         speech_text = 'I found no courses about {subject}'.format(subject=subject)
     else:
-        word = 'course about {subject}. It is' if count == 1 else 'courses about {subject}. They are'
+        word = 'course about {subject}. It is' if count == 1 else 'courses about {subject}. Here are the first two'
```
Comment (clintonb) [inline enrollments/enrollments.py]: Future improvement: If there are only two courses, use "They are ".

**Pull request merged**